src/main/java/io/jsonwebtoken/BadAudienceJwtException.java

```diff
@@ -0,0 +1,11 @@
+package io.jsonwebtoken;
+
+public class BadAudienceJwtException extends ClaimJwtException {
```

Comment:
An alternative to `expect` based on your concerns might be `require` -  `requireIssuer`, etc.  Maybe then `IncorrectClaimException` would feel better?